RocketData/DataProvider.swift

```diff
@@ -54,13 +54,56 @@ open class DataProvider<T: SimpleModel>: ConsistencyManagerListener, BatchListen
         }
     }
 
+    /**
+     This property returns the model identifier which is associated with this `DataProvider`.
+     It either returns the id of the data currently set, or an id that it's listening to (if the model doesn't exist yet).
+     If you don't have a model yet, but want to listen to a model which may exist in the future, you can set
+     this property to an ID.
+     You should NOT do this if you have already set data (this will throw an assertion).
+     After setting this property, if the model is updated elsewhere, the delegate will be called and the new model will be set.
+     */
+    open var modelIdentifier: String? {
+        get {
+            if let data = data {
+                return data.modelIdentifier
+            } else {
+                return listeningToModelIdentifier
+            }
+        }
+        set {
+            guard data == nil else {
+                Log.sharedInstance.assert(false, "You should not manually set the model identifier when you already have a model.")
+                return
+            }
+            listeningToModelIdentifier = newValue
+            if newValue != nil {
+                // If we start listening to a new ID, we need to add ourselves as a global listener
+                dataModelManager.consistencyManager.addModelUpdatesListener(self)
+            }
+        }
+    }
+
     /// This saves the batchListener instance. It is public because it implements the BatchListenable protocol. You should never edit this directly.
     open weak var batchListener: BatchDataProviderListener?
 
     // MARK: - Private instance variables
 
     /// This wraps the data with a lastUpdated ChangeTime and makes sure the data is never changed without updating the ChangeTime
-    var dataHolder = DataHolder<T?>(data: nil)
+    var dataHolder = DataHolder<T?>(data: nil) {
+        didSet {
+            if dataHolder.data != nil && listeningToModelIdentifier != nil {
+                // If we set new data, let's erase the ID we're listening to and stop listening to updates
+                listeningToModelIdentifier = nil
+                dataModelManager.consistencyManager.removeModelUpdatesListener(self)
+            }
+        }
+    }
+
+    /**
+     This is the internal storage for the ID we are listening to.
+     If `data` is not nil, this should be nil.
+     */
+    var listeningToModelIdentifier: String?
```

Comment:
It's a little weird because sometimes `data` is not nil when we set `listeningToModelIdentifier`. Since it's an internal property, this is modified right after we set data.